src/OpenTelemetry/Trace/TracerProviderSdk.cs

```diff
@@ -153,11 +164,14 @@ internal class TracerProviderSdk : TracerProvider
 
                     // Function which takes ActivitySource and returns true/false to indicate if it should be subscribed to
                     // or not.
-                    listener.ShouldListenTo = (activitySource) => regex.IsMatch(activitySource.Name);
+                    listener.ShouldListenTo = (activitySource) => string.IsNullOrEmpty(activitySource.Name) || regex.IsMatch(activitySource.Name);
```

Comment:
We dont want to affect the regular flow here by subscribing to empty sources. Do that conditionally, based of if there atleast one entry in the hashset/other datastructure keeping list of legacy activity names.

Reply:
Added the condition to check if the HashSet is not empty.